### DISCUSSION TITLE
"global" cross host subnet claim for routing / bridges

### DISCUSSION THREAD

**Discussion by DeadlyBug** [dswd/vpncloud#285] [Q&A] [answered]
Hello,
I am using VPNCloud TAP interfaces as bridges on multiple hosts, with multiple LXC containers spawned on each host.
Each container is allocated an eth interface which connects to the VPNCloud Tap bridge, meaning it can access the VPN in L2, send a DHCP request and get an IP address allocated by another "special" container which runs DNSmasq. This setup allows all containers to communicate with each other through the VPNCloud tunnel, regardless of the physical host on which the containers run. 

Host1 tap0 is 192.168.0.1
Host2 tap0 is 192.168.0.2
The DHCP container is 192.168.0.30 (fixed IP), and is allowed to allocate IP from 192.168.0.100 to 192.168.0.255 to any container requesting an IP.
br0 is created over tap0 on each host
each container is provided an eth0 interface linked to br0, providing each container access to the VPNcloud.
for example
Container1 may receive IP 192.168.0.100 from DHCP container
Container2 may receive 192.168.0.101 from DHCP container etc

What I don't understand is "_how to make the system understand that all hosts collectively claim the 192.168.0.0 **/24** subnet (or **/16** **/18** etc, we may expand the IP pool in the future)_". In the config file the "claims" option seems to be reserved for a specific host to claim a specific subnet.
On the aforementioned setup we don't know whether container1 and 2 may be hosted on which host, and we don't care. We need to make the system understand that the whole /24 subnet for 192.168.0.0 should be routed through tap0 & br0.

In peerVPN (https://peervpn.net/tutorial/) this is done easily through the ifconfig4 command which takes a subnet: `ifconfig4 192.168.0.1/24` . Can this be done as well for VPNCloud?

Thanks

**Comment by dswd** [comment, marked as answer]
Hi there,

for this use case you need to use TAP mode (as your are doing right now). In this mode, VpnCloud does not care about IPs at all, it works on the level of MAC addresses and lets the Linux kernel do the IP stuff.
The "claims" option in the config file is used in TUN mode and does not apply to your case.
You should be able to use TAP mode and configure your subnet using the "ip" option like this: "ip: 1192.168.0.1/24".

**Comment by DeadlyBug** [comment]
Perfect, many thanks.
Maybe the documentation (or default conf file) could be updated to include this bit?
